src/sentry/api/endpoints/organization_member/index.py

```diff
@@ -376,28 +376,32 @@ def post(self, request: Request, organization) -> Response:
         members_can_only_invite_to_members_teams = (
             not organization.flags.allow_joinleave and not organization.flags.disable_member_invite
         )
-        has_team_roles = "teamRoles" in result and bool(result["teamRoles"])
+        has_teams = bool(
+            ("teamRoles" in result and result["teamRoles"])
+            or ("teams" in result and result["teams"])
+        )
 
-        if is_member and members_can_only_invite_to_members_teams and has_team_roles:
+        if is_member and members_can_only_invite_to_members_teams and has_teams:
             requester_teams = set(
                 OrganizationMember.objects.filter(
                     organization=organization,
                     user_id=request.user.id,
                     user_is_active=True,
                 ).values_list("teams__slug", flat=True)
             )
-            team_slugs = [team.slug for team, _ in result.get("teamRoles")]
+            team_slugs = teams = (
+                [team.slug for team, _ in result.get("teamRoles")]
+                if "teamRoles" in result and result["teamRoles"]
+                else [team.slug for team in result.get("teams")]
+            )
```

Comment:
i _think_ you can simplify to:
```suggestion
            team_slugs = [team.slug for team, _ in result.get("teamRoles", [])] or [team.slug for team in result.get("teams", [])]
```